### PR TITLE
fix(feed-item-row): spacing and transition improvements - Vue 2

### PR DESCRIPTION
**Pull request by braddialpad**
# fix(feed-item-row): spacing and transition improvements

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [x] Fix
- [ ] Feature
- [ ] Performance Improvement
- [ ] Refactor

These types will not increment the version number, but will still deploy to documentation site on release:

- [ ] Documentation
- [ ] Build system
- [ ] CI
- [ ] Style (code style changes, not css changes)
- [ ] Test
- [ ] Other (chore)

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1185

## :book: Description

A bunch of fixes requested by the design team from the feed-item-row private beta. Mostly related to spacing. 

- Fixed the transition which was not working as intended.
- Converted all utility classes to tokens.
- Had to override a bunch of stuff from dt-item-layout (padding, margin etc) that we probably shouldn't have to, but fixing that properly is outside the scope of this. This was the reason a lot of spacing was off in the first place.

## :bulb: Context

Have the feed-item-row component match the designs, there were some spacing issues.

Confirmed with Sanjeev that the transition was only supposed to be happening on state change (searched, error) not on isActive change. So this has been fixed by only enabling the transition in these cases.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all CSS changes:

- [x] I have used design tokens whenever possible.
- [x] I have considered how this change will behave on different screen sizes.
- [x] I have visually validated my change in light and dark mode.
- [x] I have used gap or flexbox properties for layout instead of margin whenever possible.

## :link: Sources

Dialpad thread: https://dialpad.com/app/messages/agxzfnViZXItdm9pY2VyGAsSC1RleHRNZXNzYWdlGICA4d_i9qgIDA

### CODE REVIEW

File: packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue [hunk 1]
```diff
@@ -271,52 +286,67 @@ export default {
...
-      text-overflow: ellipsis;
-      overflow: hidden;
-      white-space: nowrap;
     }
 
     &__time {
       font-size: var(--dt-font-size-100);
-      margin-top: var(--dt-space-200);
-      line-height: var(--dt-font-line-height-300);
+      line-height: 2.1rem;
```
Comment (braddialpad) [inline packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue]: hardcoded line height here since the time uses a different font size than the field it is aligning to, and they need to have the same line height to center align correctly.

File: packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue [hunk 2]
```diff
@@ -246,20 +251,30 @@ export default {
...
+      }
+    },
+  },
+
   methods: {
+    transitionComplete () {
+      if (this.state === DEFAULT_FEED_ROW_STATE) {
+        this.transitionActive = false;
+      }
+    },
```
Comment (braddialpad) [inline packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue:279]: Have to do this or it will cut out the transition instead of waiting for it to finish.

File: packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row_default.story.vue [hunk 3]
```diff
@@ -1,66 +1,66 @@
 <template>
-  <ul class="d-pt16 feed-item-row__default-story">
```
Comment (braddialpad) [inline packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row_default.story.vue:2]: removed this ul and put in decorators since it's not part of the component.

File: packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row_default.story.vue [hunk 4]
```diff
@@ -159,29 +159,3 @@ export default {
...
-    &__reply{
-      color: var(--dt-color-foreground-tertiary);
-      font-size: 12px;
-      font-style: normal;
-      font-weight: 400;
-      line-height: 16px;
-      letter-spacing: -0.12px;
-    }
-  }
-</style>
```
Comment (braddialpad) [inline packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row_default.story.vue:187]: All this styling on the example makes it very confusing to know what the component is doing and what the example is doing.. removed.

File: packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue (hunk 1 above)
Comment (juliodialpad) [reply packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue]: This will raise a linter warning isn't it?

**Comment by juliodialpad** [comment]
The `Click to fade` button is not working on Vue 2

Screenshot 2024-01-18 at 1 22 34 p m

File: packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue (hunk 1 above)
Comment (braddialpad) [reply packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue]: nah only if it's px rather than rem

**Review by juliodialpad — COMMENTED** [review]
Changes looking good, when you click to fade it's supposed to change from yellow to transparent, right? 

https://github.com/dialpad/dialtone/assets/87546543/39a7f99c-e0d8-4105-8d0a-5e1f75b89dbe

**Comment by braddialpad** [comment]
> The `Click to fade` button is not working on Vue 2

Ah thanks

File: packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue (hunk 1 above)
Comment (juliodialpad) [reply packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue]: Cool, thought it was a warning on any fixed value instead of CSS var.

**Comment by braddialpad** [comment]
> when you click to fade it's supposed to change from yellow to transparent, right?

Since isActive is enabled on that it should actually fade to the active state. It does that now after my change. May have not been working correctly before.

File: packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue (hunk 1 above)
Comment (francisrupert) [reply packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue]: @braddialpad If the parent is `display: flex`, I'm guessing it is `align-items: center` and can understand your `line-height` workaround. Would `align-items: baseline` work instead?

Comment (braddialpad) [reply packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue]: hmm I had some issues with it previously but let me try again after my recent changes.

**Review by juliodialpad — APPROVED** [review]
Looking good on Vue 2, are you going to create another PR for Vue 3 changes, right?

File: packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue (hunk 1 above)
Comment (braddialpad) [reply packages/dialtone-vue2/recipes/conversation_view/feed_item_row/feed_item_row.vue]: Did a bit of reconfiguring and it works well with baseline now, Looks pretty good!

**Comment by braddialpad** [comment]
> Looking good on Vue 2, are you going to create another PR for Vue 3 changes, right?

Yep will be a separate PR

**Comment by braddialpad** [comment]
> SyntaxError: ':focus-visible' is not a valid selector

Hmm some jsdom issue.. strange it doesn't happen locally.

**Comment by braddialpad** [comment]
> > SyntaxError: ':focus-visible' is not a valid selector
> 
> Hmm some jsdom issue.. strange it doesn't happen locally.

Annnnddd it just passed on the 2nd run...

**Comment by francisrupert** [comment]
@braddialpad Will [Feed Item Reply Bar](https://www.figma.com/file/oTY56LgQha8ljb4c8w4GtO/Dialpad-App-Components?type=design&node-id=9240-7140&mode=design&t=gXTuKTRLjrYAuqKk-11) be its own recipe, or should it be part of this?

image

**Comment by braddialpad** [comment]
> @braddialpad Will [Feed Item Reply Bar](https://www.figma.com/file/oTY56LgQha8ljb4c8w4GtO/Dialpad-App-Components?type=design&node-id=9240-7140&mode=design&t=gXTuKTRLjrYAuqKk-11) be its own recipe, or should it be part of this?

Yep completely different recipe that hasn't been done yet. What's in there is just a placeholder for the example. The one that's currently on DP 1.0 in product looks pretty much identical to that and we just pass it into the slot for now.
